platform/src/server/http/http_server.ts

```diff
@@ -55,41 +86,79 @@ export class HttpServer {
       return;
     }
 
-    await promisify(this.server.close).call(this.server);
-
+    await this.server.stop();
     this.server = undefined;
   }
 
   private initializeServer(config: HttpConfig) {
-    // TODO This was forced to `any` because of the following problem with the
-    // express declaration file, and can be removed once that is fixed:
-    //
-    // ```
-    // Argument of type 'Application' is not assignable to parameter of type
-    // '((request: IncomingMessage, response: ServerResponse) => void) | undefined'
-    // ```
-    const app = this.app as any;
-
-    if (!config.ssl.enabled) {
-      return http.createServer(app);
-    }
+    const options: ServerOptions = {
+      host: config.host,
+      port: config.port,
+      routes: {
+        cors: config.cors,
+        payload: {
+          maxBytes: config.maxPayload.getValueInBytes(),
+        },
+        validate: {
+          options: {
+            abortEarly: false
+          }
+        }
+      },
+      state: {
+        strictHeader: false
```

Comment:
I thought we had strictHeader enabled on the old server. Did you check that?

Reply:
Here is what we have in master: https://github.com/elastic/kibana/blob/97df91f36f991cc38cdd0d911618a5bba7da52ac/src/server/http/setup_connection.js#L11-L13 - `strictHeader: false`.